.github/workflows/post_release_version_bump.yml

```diff
@@ -0,0 +1,136 @@
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_ARN_SECRETS_MANAGER }}
+          aws-region: ${{ env.AWS_DEFAULT_REGION }}
+
+      - name: Get Bot secrets
+        uses: aws-actions/aws-secretsmanager-get-secrets@v1
+        id: bot_secrets
+        with:
+          secret-ids: |
+            BOT_TOKEN ,${{ secrets.BOT_TOKEN_SECRET_ARN }}
+          parse-json-secrets: true
+
+      - name: Setup Git
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          token: ${{ env.BOT_TOKEN_GITHUB_RW_PATOKEN }}
+
+      - name: Configure Git
+        run: |
+          git config user.name "github-actions"
+          git config user.email "github-actions@github.com"
+
+      - name: Extract Major.Minor Version and setup Env variable
+        run: |
+          echo "VERSION=${{ github.event.inputs.version }}" >> $GITHUB_ENV
+          echo "MAJOR_MINOR=$(echo ${{ github.event.inputs.version }} | sed -E 's/([0-9]+\.[0-9]+)\.[0-9]+/\1/')" >> $GITHUB_ENV
+
+      - name: Determine release branch and checkout
+        run: |
+          RELEASE_BRANCH="release/v${MAJOR_MINOR}.x"
+          git fetch origin $RELEASE_BRANCH
+          git checkout -b "prepare-main-for-next-dev-cycle-${VERSION}" origin/$RELEASE_BRANCH
+
+      - name: Update version to next development version in main
+        run: |
+          DEV_VERSION="${{ github.event.inputs.version }}.dev0"
+          sed -i "s/public final String VERSION = \".*\";/public final String VERSION = \"${DEV_VERSION}\";/" awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/ArtifactVersion.java
+          VERSION="${{ github.event.inputs.version }}"
+          git add awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/ArtifactVersion.java
+
+      - name: Append latest release checksum to release-artifacts-metadata.json
+        run: |
+          ARTIFACT_NAME="aws-opentelemetry-agent.jar"
+          curl -L -o $ARTIFACT_NAME https://github.com/aws-observability/aws-otel-java-instrumentation/releases/download/v${{ github.event.inputs.version }}/$ARTIFACT_NAME
+
+          CHECKSUM=$(shasum -a 256 $ARTIFACT_NAME | awk '{ print $1 }')
+
+          FILE="release-artifacts-metadata.json"
+          UPDATED_JSON=$(jq --arg version "${{ env.VERSION }}" \
+                            --arg name "$ARTIFACT_NAME" \
+                            --arg checksum "$CHECKSUM" \
+                            '.release += [{"version": $version, "checksum": [{"name": $name, "checksum": $checksum}]}]' "$FILE")
+
+          echo "$UPDATED_JSON" > "$FILE"
+          git add release-artifacts-metadata.json
+
+      - name: Push changes to Github
+        run: |
+          git commit -m "Prepare main for next development cycle: Update version to ${{ github.event.inputs.version }}"
+          git push --set-upstream origin "prepare-main-for-next-dev-cycle-${{ github.event.inputs.version }}"
```

Comment:
Should use `{VERSION}` or `{{ env.VERSION }}` everywhere once it's available, not the input directly